### PR TITLE
Utils: Use 's' as default for unit-less intervals

### LINKED ISSUE #22362: Explore doesn't honour Prometheus step
In this explore panel, we can see the http request using step 15 even though I've put both in at 60.

Screenshot 2020-02-21 at 09 49 18

### PR DESCRIPTION
**Pull request by s-h-a-d-o-w**
**What this PR does / why we need it**:
If the user specifies a string that is a unit-less number, it is assumed that they meant seconds.
Resolves the problem that users don't get feedback if they input a unit-less number e.g. in Explore. This will currently affect prometheus and cloudwatch data sources.

**Which issue(s) this PR fixes**:
Fixes #22362

### CODE REVIEW

File: public/app/plugins/datasource/prometheus/components/PromExploreQueryEditor.tsx [hunk 1]
```diff
@@ -48,7 +48,7 @@ export function PromExploreQueryEditor(props: Props) {
           onKeyDownFunc={onReturnKeyDown}
           value={query.interval || ''}
           hasTooltip={true}
-          tooltipContent={'Needs to be a valid time unit string, for example 5s, 1m, 3h, 1d, 1y'}
+          tooltipContent={'Time units can be used here, e.g.: 5s, 1m, 3h, 1d, 1y (Default if no unit is specified: s)'}
```
Comment (Estrax) [inline public/app/plugins/datasource/prometheus/components/PromExploreQueryEditor.tsx:51]: Can we change the tooltip content sentence so it (visually) looks better? 
Screenshot from 2020-04-01 11-43-33

**Review by Estrax — APPROVED** [review]
LGTM. Only one small comment below.

**Comment by Estrax** [comment]
@s-h-a-d-o-w you probably need to update tests & snapshots too. Run `yarn jest -u` to update snapshots and look at `public/app/features/variables/interval/action.test.ts`.

File: public/app/plugins/datasource/prometheus/components/PromExploreQueryEditor.tsx (hunk 1 above)
Comment (s-h-a-d-o-w) [reply public/app/plugins/datasource/prometheus/components/PromExploreQueryEditor.tsx:51]: Do you have a suggestion? (`FormLabel` only accepts string for tooltip, so I can't insert a line break before the parens.)

File: public/app/core/utils/kbn.ts [hunk 2]
```diff
@@ -200,9 +200,22 @@ kbn.calculateInterval = (range: TimeRange, resolution: number, lowLimitInterval:
 };
 
 kbn.describe_interval = (str: string) => {
+  // Default to seconds if no unit is provided
+  if (Number(str)) {
+    return {
+      sec: kbn.intervals_in_seconds.s,
+      type: 's',
+      count: parseInt(str, 10),
```
Comment (aocenas) [inline public/app/core/utils/kbn.ts:208]: Don't we need floating point here? Not sure about the `interval_regex` seems like it allows floating point but then we also use `parseInt`, but just wanted to check to be sure.

**Review by aocenas — APPROVED** [review]
Good to go, just small thing to check, if thats fine can be merged.

File: public/app/core/utils/kbn.ts (hunk 2 above)
Comment (s-h-a-d-o-w) [reply public/app/core/utils/kbn.ts:208]: That's true. I was going by the `parseInt` that was already there. I suppose those two should be harmonized - either the regex changed or using `parseFloat` instead.

So - do we want to allow the user to be able to enter floating point values?

Comment (aocenas) [reply public/app/core/utils/kbn.ts:208]: Well I just wanted to make sure this is the same but I did not run it so I was not sure. It should definitely be unified if it isn't at the moment, probably in alignment with the old behavior. So if you can specify `0.01s` we should also allow `0.01` unitless input and treat is as `0.01s`.

Comment (s-h-a-d-o-w) [reply public/app/core/utils/kbn.ts:208]: But the old behavior already conflicted. I didn't change either the regex, nor the original `parseInt`.
The part I added works precisely like the original code.
If you enter `1.5m` in our cluster right now, you'll see that it results in step `60`, even though it should obviously be `90` if floating point was working.

Maybe it was intentional to permit floating point values in principle but to then silently drop the part after the decimal point?

Comment (s-h-a-d-o-w) [reply public/app/core/utils/kbn.ts:208]: Just spoke about this with @aocenas , we agreed that the previous behavior will remain untouched and I should just ensure that when numbers without units are entered, it works the same way. Which it does, so this is good to go. :rocket:

**Pull request merged**